Emby.Server.Implementations/Session/SessionWebSocketListener.cs

```diff
@@ -99,5 +140,205 @@ private void EnsureController(SessionInfo session, IWebSocketConnection connecti
             var controller = (WebSocketController)controllerInfo.Item1;
             controller.AddWebSocket(connection);
         }
+
+        /// <summary>
+        /// Called when a WebSocket is closed.
+        /// </summary>
+        /// <param name="sender">The WebSocket.</param>
+        /// <param name="e">The event arguments.</param>
+        private void OnWebSocketClosed(object sender, EventArgs e)
+        {
+            var webSocket = (IWebSocketConnection) sender;
+            _logger.LogDebug("WebSocket {0} is closed.", webSocket);
+            RemoveWebSocket(webSocket);
+        }
+
+        /// <summary>
+        /// Adds a WebSocket to the KeepAlive watchlist.
+        /// </summary>
+        /// <param name="webSocket">The WebSocket to monitor.</param>
+        private void KeepAliveWebSocket(IWebSocketConnection webSocket)
```

Comment:
It was, but it got changed, amongst other reasons, because by `.Wait()`-ing there we can catch a possible exception, instead of letting it slip with the async task.

Another reason is that Sonar complained about not awaiting for the async task when calling `KeepAliveWebSocket` from `OnServerManagerWebSocketConnected` which is not `async` as well. And this issue was seen as a `Major` code smell.
```
Because this call is not awaited, execution of the current method continues before the call is completed.
Consider applying the 'await' operator to the result of the call.
```

Reply:
`OnServerManagerWebSocketConnected` can be made async

Reply:
This solves my dilemma. I'll make it async then.